pandas/core/frame.py

```diff
@@ -912,6 +916,17 @@ def __init__(
 
         NDFrame.__init__(self, mgr)
 
+        if original_dtype is None and is_pandas_object and data_dtype == np.object_:
```

Comment:
do you want to do anything with dict_of_pandas_objects?

Reply:
That is the funny thing, they already preserve the dtype...